galpy/orbit/Orbits.py

```diff
@@ -5037,34 +5036,30 @@ def animate(self,*args,**kwargs): #pragma: no cover
 """.format(trace_indx=str(ii),trace_num_1=str(2*ii+1),trace_num_2=str(2*ii+2),
            line_color=line_colors[ii])
             traces_cumul+= """,trace{trace_num_1},trace{trace_num_2}""".format(trace_num_1=str(2*ii+1),trace_num_2=str(2*ii+2))
-        update_trace12= """
-      Plotly.extendTraces('{divid}', {{
-        x: [data.x1_0.slice(trace_slice_begin,trace_slice_end)],
-        y: [data.y1_0.slice(trace_slice_begin,trace_slice_end)],
-      }}, [0]);
-      trace_slice_begin-= trace_slice_len;
-      trace2= {{
-        x: [data.x1_0.slice(trace_slice_begin,trace_slice_end)], 
-        y: [data.y1_0.slice(trace_slice_begin,trace_slice_end)],
-      }};
-      Plotly.restyle('{divid}',trace2,[1]);
-""".format(divid=self.divid)
-        for ii in range(1,self.size):
-            update_trace12+= """
-      trace_slice_begin+= trace_slice_len;
-      Plotly.extendTraces('{divid}', {{
-        x: [data.x1_{trace_indx}.slice(trace_slice_begin,trace_slice_end)],
-        y: [data.y1_{trace_indx}.slice(trace_slice_begin,trace_slice_end)],
-      }}, [{trace_num_10}]);
-      trace_slice_begin-= trace_slice_len;
-      trace{trace_num_2}= {{
-        x: [data.x1_{trace_indx}.slice(trace_slice_begin,trace_slice_end)], 
-        y: [data.y1_{trace_indx}.slice(trace_slice_begin,trace_slice_end)],
-      }};
-      Plotly.restyle('{divid}',trace{trace_num_2},[{trace_num_20}]);
-""".format(divid=self.divid,trace_indx=str(ii),trace_num_1=str(2*ii+1),
-           trace_num_2=str(2*ii+2),trace_num_10=str(2*ii+1-1),
-           trace_num_20=str(2*ii+2-1))
+        x_data_list = """"""
+        y_data_list = """"""
+        trace_num_10_list = """"""
+        trace_num_20_list = """"""
+        for jj in range(len(d1s)):
+            for ii in range(0, self.size):
+                x_data_list += """data.x{jj}_{trace_indx}.slice(trace_slice_begin,trace_slice_end), """.format(jj=jj+1,
+                    divid=self.divid, trace_indx=str(ii))
+                y_data_list += """data.y{jj}_{trace_indx}.slice(trace_slice_begin,trace_slice_end), """.format(jj=jj+1,
+                    divid=self.divid, trace_indx=str(ii))
+                trace_num_10_list += """{trace_num_10}, """.format(trace_num_10=str(2*jj*self.size + 2 * ii + 1 - 1))
+                trace_num_20_list += """{trace_num_20}, """.format(trace_num_20=str(2*jj*self.size + 2 * ii + 2 - 1))
+
+        master_traces = """
+        traces = {{x: [{x_data_list}], y: [{y_data_list}]
+        }}
+        """.format(x_data_list=x_data_list, y_data_list=y_data_list)
+        update_trace12 = """
+        {master_traces}
+        trace_slice_begin+= trace_slice_len;
```

Comment:
I don't think this should be here (because in the original, this was as part of the second trace to extend, to make up for the ``trace_slice_begin-= trace_slice_len;`` just before the ``Plotly.restyle`` of the first trace. Not sure how the code actually works in this case though, because I think the trace you are extending should have zero length...